### PR TITLE
feat(messaging): add Signal adapter via signal-cli JSON-RPC daemon

### PR DESCRIPTION
**Pull request by ibhagwan**
Implements Signal messaging support using the signal-cli daemon HTTP API,
following the existing adapter architecture (Telegram, Discord, Slack, Twitch).

- Inbound: SSE stream with automatic reconnection and exponential backoff
  (2s → 60s), UTF-8 chunk boundary handling, buffer overflow protection.
- Outbound: JSON-RPC send calls. DM recipients must be a JSON array.
- Typing indicators: JSON-RPC sendTyping with ~5s expiry.
- Attachments: Temp files in {instance_dir}/tmp/, auto-cleaned after send.
- Streaming: Not supported (Signal can't edit messages).
- Permissions: DM allowlist + group filter (None = block all groups).

Config types in types.rs, toml_schema.rs, load.rs. SignalPermissions in
permissions.rs with from_config/from_instance_config. Hot-reload support in
watcher.rs. 23 unit tests.

Add to config.toml:
```toml
[messaging.signal]
enabled = true
http_url = "http://127.0.0.1:8686/"
account = "+1234567890"
dm_allowed_users = ["+0987654321"]
group_ids = ["<uuid1>", "<uuid2>"]
group_allowed_users = ["+5566778899", "+1122334455"]
ignore_stories = true

[[messaging.signal.instances]]
name = "work"
enabled = true
http_url = "http://127.0.0.1:8687/"
account = "+1122334455"
dm_allowed_users = ["+5566778899"]
group_ids = ["<uuid1>", "<uuid2>"]
group_allowed_users = ["+1122334455"]
```

Requires signal-cli daemon running: `signal-cli daemon --http`

Closes https://github.com/spacedriveapp/spacebot/issues/310

> [!NOTE]
> **Summary:** Complete Signal adapter implementation (1953 lines) following the existing messaging adapter pattern. Adds SSE-based inbound stream with exponential backoff reconnection, JSON-RPC outbound messaging, typing indicators, file attachment handling, and fine-grained permission controls (DM allowlist + group filters). Configuration schema extended for Signal with multi-instance support. Includes 23 unit tests covering JSON parsing, permissions, and SSE edge cases. No changes to core agent or Rig integration—purely additive messaging module.
>
> <sub>Written by [Tembo](https://app.tembo.io) for commit [83b8f72](https://github.com/spacedriveapp/spacebot/commit/83b8f725d2ff4cc42b314418500126f25e8fc8e4). This will update automatically on new commits.</sub>

### CODE REVIEW

**Comment by jamiepine** [comment]
Woah! Thank you, huge.

**Comment by ibhagwan** [comment]
Ty @jamiepine for this wonderful project, although lesser known ATM I believe this one has huge potential, I've been personally testing every decent project out there and so far I like this one and IronClaw the most.

Btw, this isn't my first signal support PR, I've added the same to ZeroClaw, IronClaw, NullClaw and Hermes-agent:
https://github.com/zeroclaw-labs/zeroclaw/pull/468
https://github.com/nearai/ironclaw/pull/271
https://github.com/nullclaw/nullclaw/pull/63
https://github.com/NousResearch/hermes-agent/pull/268

**Comment by ibhagwan** [comment]
This needs some work, I'll change it to draft until I respond to all the review comments

**Comment by jamiepine** [comment]
> Ty @jamiepine for this wonderful project, although lesser known ATM I believe this one has huge potential, I've been personally testing every decent project out there and so far I like this one and IronClaw the most.
> 
> Btw, this isn't my first signal support PR, I've added the same to ZeroClaw, IronClaw, NullClaw and Hermes-agent: [zeroclaw-labs/zeroclaw#468](https://github.com/zeroclaw-labs/zeroclaw/pull/468) [nearai/ironclaw#271](https://github.com/nearai/ironclaw/pull/271) [nullclaw/nullclaw#63](https://github.com/nullclaw/nullclaw/pull/63) [NousResearch/hermes-agent#268](https://github.com/NousResearch/hermes-agent/pull/268)

This is wonderful to hear, thank you. I'm putting a load of effort into making this the best I can. Hoping soon it will pick up traction wise, but so far the contributions have been amazing and we're moving quickly to stability. If you want to chat please join the Discord and/or DM me anytime!

https://discord.gg/g8fMZsVSTT

**Comment by ibhagwan** [comment]
@coderabbitai review

**Comment by ibhagwan** [comment]
@coderabbitai review

**Comment by ibhagwan** [comment]
@coderabbitai review

**Comment by ibhagwan** [comment]
@coderabbitai review

**Comment by ibhagwan** [comment]
@jamiepine, addressed everything by coderabbit/tembo, fmt, clippy and signal tests pass (some local test failures unrelated to the code, I believe will pass in the CI), I think this is ready

**Comment by ibhagwan** [comment]
@coderabbitai review

**Comment by ibhagwan** [comment]
@coderabbitai review

File: src/config/toml_schema.rs [hunk 1]
```diff
@@ -491,6 +491,7 @@ pub(super) struct TomlMessagingConfig {
     pub(super) email: Option<TomlEmailConfig>,
     pub(super) webhook: Option<TomlWebhookConfig>,
     pub(super) twitch: Option<TomlTwitchConfig>,
+    pub(super) signal: Option<TomlSignalConfig>,
 }
```
[the remark this answers is not in the document]
Comment (ibhagwan) [reply src/config/toml_schema.rs:500]: we don't need this in the binding

**Comment by ibhagwan** [comment]
@coderabbitai review

**Comment by ibhagwan** [comment]
@coderabbitai review

**Comment by ibhagwan** [comment]
@coderabbitai review

**Comment by ibhagwan** [comment]
@coderabbitai review
